plugins/reporters/cyclonedx/src/main/kotlin/CycloneDxReporter.kt

```diff
@@ -356,17 +358,16 @@ class CycloneDxReporter : Reporter {
         schemaVersion: Version,
         outputDir: File,
         outputName: String,
-        requestedOutputFileFormats: Set<FileFormat>
+        validFileExtensions: List<String>
     ): List<File> {
         val writtenFiles = mutableListOf<File>()
-        val outputFileFormats = supportedOutputFileFormats.filter { it in requestedOutputFileFormats }
 
-        outputFileFormats.forEach { fileFormat ->
-            val outputFile = outputDir.resolve("$outputName.${fileFormat.fileExtension}")
+        validFileExtensions.forEach { fileExtension ->
+            val outputFile = outputDir.resolve("$outputName.$fileExtension")
 
-            val bomGenerator = when (fileFormat) {
-                FileFormat.XML -> BomGeneratorFactory.createXml(schemaVersion, bom)
-                FileFormat.JSON -> {
+            val bomGenerator = when (fileExtension) {
+                "xml" -> BomGeneratorFactory.createXml(schemaVersion, bom)
+                "json" -> {
```

Comment:
IIUC this change is a functional change at least in the following aspects
1. If "unkown" file extension get specified, these get ignored. Before it failed.
2. The default is changed from xml to all formats.

Looking at this when case, the previous code did not redefine constants and had compiler check
whether all cases are covered by `when`.  What do you think about "reverting" that by simply defing a file extensions enum? (And keeping the code analog to before)

Reply:
I was considering to introduce a separate file format enum for CycloneDX here, but then felt that 1) this case is so simple that it barely justifies the additional code, 2) that functionality should actually go upstream (I'm planning to make a contribution to the CycloneDX Java library so it "known" about its own formats that it supports).

Regarding the behavioral changes, you're correct. I'll change it back to fail again on an unknown format, and to default only to XML.

Reply:
> so simple that it barely justifies the additional code

I was proposing the enum, because I thought it would actually safe code:
e.g. you could turn the input string into enum instance via clikt (as before) instead of adding logic which checks for valid values.

Reply:
> you could turn the input string into enum instance via clikt (as before)

Note that clikt was / is not involved here, as the reporters only see their reporter-specific options as part of the generic `PluginConfiguration`, and the `ReporterCommand` does not know about typed reporter-specific options.

Reply:
> I'm planning to make a contribution to the CycloneDX Java library so it "known" about its own formats that it supports

FYI: https://github.com/CycloneDX/cyclonedx-core-java/pull/445